sufia-models/lib/generators/sufia/models/geonames_username_config_generator.rb

```diff
@@ -0,0 +1,21 @@
+require 'rails/generators'
+
+class Sufia::Models::GeonamesUsernameConfigGenerator < Rails::Generators::Base
+  source_root File.expand_path('../templates', __FILE__)
+
+  desc """
+This Generator makes the following changes to your application:
+  1. Updates existing sufia.rb initializer to include a geonames_username configuration
+       """
+
+  def banner
+    say_status("info", "ADDING GEONAMES_USERNAME OPTION TO SUFIA CONFIG", :blue)
+  end
+
+  def inject_config_initializer
+    inject_into_file 'config/initializers/sufia.rb', before: "# Specifies a user account for the Geonames API (please override this value)" do
```

Comment:
I think you want the "# Specifies a user account..." string on the next line, and you want an existing string in the Sufia config here. Like:

```ruby
inject_into_file 'config/initializers/sufia.rb', before: "# Where to store tempfiles, leave blank for the system temp  directory (e.g. /tmp)" do
  "# Specifies a user account for the Geonames API (please override this value).\n" +
  "# Default is empty string\n" +
  "config.geonames_username = ''\n"
end
```